mod_wsbridge/mod_wsbridge.c

```diff
@@ -101,6 +104,41 @@ switch_endpoint_interface_t *wsbridge_endpoint_interface;
 static switch_memory_pool_t *module_pool = NULL;
 static int running = 1;
 
+struct Queue {
+	switch_queue_t *queue;
+	switch_mutex_t *mutex;
+	unsigned int count;
+	unsigned int size;
+};
+
+void Queue_create(struct Queue *q, unsigned int capacity, switch_memory_pool_t *memory_pool) {
+	q->count = 0;
+	q->size = capacity;
+	switch_queue_create(&q->queue, q->size, memory_pool);
+	switch_mutex_init(&q->mutex, SWITCH_MUTEX_NESTED, memory_pool);
+}
+
+switch_status_t Queue_push(struct Queue *q, void *data) {
+	switch_status_t rv = SWITCH_STATUS_FALSE;
+	switch_mutex_lock(q->mutex);
+	if ((q->count < q->size) && ((rv = switch_queue_trypush(q->queue, data)) == SWITCH_STATUS_SUCCESS)) {
+		q->count++;
+	}
```

Comment:
remove checks from count and size and leave it as
`rv = switch_queue_trypush(q->queue, data);`